miner/scroll_worker.go

```diff
@@ -579,11 +583,19 @@ func (w *worker) startNewPipeline(timestamp int64) {
 			return
 		}
 	}
+
+	// pipelineCCC was nil, so the block was built for RPC purposes only. Stop the pipeline immediately
+	// and update the pending block.
+	if pipelineCCC == nil {
```

Comment:
Why do we need this? So that we don't wait for the timeout?

Reply:
Yeah, exactly. If we do wait for the timeout, it is highly likely the actual block will be gossiped by the sequencer before we can finalize ours.